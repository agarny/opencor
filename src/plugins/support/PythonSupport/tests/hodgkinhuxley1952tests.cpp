/*******************************************************************************

Copyright (C) The University of Auckland

OpenCOR is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

OpenCOR is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see <https://gnu.org/licenses>.

*******************************************************************************/

//==============================================================================
// Python support Hodgkin-Huxley 1952 tests
//==============================================================================

#include "../../../../tests/src/testsutils.h"

//==============================================================================

#include "hodgkinhuxley1952tests.h"

//==============================================================================

#include <QtTest/QtTest>

//==============================================================================

void HodgkinHuxley1952Tests::tests()
{
    // Some tests to make sure that the Hodgkinâ€“Huxley 1952 model works fine

    QStringList output;

    QVERIFY(!OpenCOR::runCli({ "-c", "PythonShell", OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/hodgkinhuxley1952tests.py") }, output));
    QCOMPARE(output, OpenCOR::fileContents(OpenCOR::fileName("src/plugins/support/PythonSupport/tests/data/hodgkinhuxley1952tests.out")));
}

//==============================================================================

QTEST_APPLESS_MAIN(HodgkinHuxley1952Tests)

//==============================================================================
// End of file
//==============================================================================
