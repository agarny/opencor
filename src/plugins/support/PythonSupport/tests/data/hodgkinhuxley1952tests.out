---------------------------------------
       Hodgkin-Huxley 1952 model
---------------------------------------
 - CVODE:
    - Constants:
       - membrane/E_R = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - membrane/Cm = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
       - sodium_channel/g_Na = [ 120.0, 120.0, 120.0, ..., 120.0, 120.0, 120.0 ]
       - potassium_channel/g_K = [ 36.0, 36.0, 36.0, ..., 36.0, 36.0, 36.0 ]
       - leakage_current/g_L = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel/E_Na = [ -115.0, -115.0, -115.0, ..., -115.0, -115.0, -115.0 ]
       - potassium_channel/E_K = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - leakage_current/E_L = [ -10.6, -10.6, -10.6, ..., -10.6, -10.6, -10.6 ]
    - States:
       - membrane/V = [ 0.0, 0.3, 0.4, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/sodium_channel_h_gate/h = [ 0.6, 0.6, 0.6, ..., 0.6, 0.6, 0.6 ]
       - potassium_channel/potassium_channel_n_gate/n = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
    - Rates:
       - membrane/V/prime = [ 0.6, 0.2, 0.1, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - membrane/i_Stim = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/alpha_m = [ 0.2, 0.2, 0.2, ..., 0.2, 0.2, 0.2 ]
       - sodium_channel/sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/potassium_channel_n_gate/alpha_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/i_Na = [ 1.0, 1.1, 1.1, ..., 1.2, 1.2, 1.2 ]
       - sodium_channel/sodium_channel_m_gate/beta_m = [ 4.0, 4.1, 4.1, ..., 4.0, 4.0, 4.0 ]
       - sodium_channel/sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/beta_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/i_K = [ -4.8, -4.6, -4.4, ..., -4.4, -4.4, -4.4 ]
       - leakage_current/i_L = [ 3.2, 3.3, 3.3, ..., 3.2, 3.2, 3.2 ]
 - Euler (forward):
    - Constants:
       - membrane/E_R = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - membrane/Cm = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
       - sodium_channel/g_Na = [ 120.0, 120.0, 120.0, ..., 120.0, 120.0, 120.0 ]
       - potassium_channel/g_K = [ 36.0, 36.0, 36.0, ..., 36.0, 36.0, 36.0 ]
       - leakage_current/g_L = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel/E_Na = [ -115.0, -115.0, -115.0, ..., -115.0, -115.0, -115.0 ]
       - potassium_channel/E_K = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - leakage_current/E_L = [ -10.6, -10.6, -10.6, ..., -10.6, -10.6, -10.6 ]
    - States:
       - membrane/V = [ 0.0, 0.3, 0.4, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/sodium_channel_h_gate/h = [ 0.6, 0.6, 0.6, ..., 0.6, 0.6, 0.6 ]
       - potassium_channel/potassium_channel_n_gate/n = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
    - Rates:
       - membrane/V/prime = [ 0.6, 0.2, 0.1, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - membrane/i_Stim = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/alpha_m = [ 0.2, 0.2, 0.2, ..., 0.2, 0.2, 0.2 ]
       - sodium_channel/sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/potassium_channel_n_gate/alpha_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/i_Na = [ 1.0, 1.1, 1.1, ..., 1.2, 1.2, 1.2 ]
       - sodium_channel/sodium_channel_m_gate/beta_m = [ 4.0, 4.1, 4.1, ..., 4.0, 4.0, 4.0 ]
       - sodium_channel/sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/beta_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/i_K = [ -4.8, -4.6, -4.4, ..., -4.4, -4.4, -4.4 ]
       - leakage_current/i_L = [ 3.2, 3.3, 3.3, ..., 3.2, 3.2, 3.2 ]
 - Heun:
    - Constants:
       - membrane/E_R = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - membrane/Cm = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
       - sodium_channel/g_Na = [ 120.0, 120.0, 120.0, ..., 120.0, 120.0, 120.0 ]
       - potassium_channel/g_K = [ 36.0, 36.0, 36.0, ..., 36.0, 36.0, 36.0 ]
       - leakage_current/g_L = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel/E_Na = [ -115.0, -115.0, -115.0, ..., -115.0, -115.0, -115.0 ]
       - potassium_channel/E_K = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - leakage_current/E_L = [ -10.6, -10.6, -10.6, ..., -10.6, -10.6, -10.6 ]
    - States:
       - membrane/V = [ 0.0, 0.3, 0.4, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/sodium_channel_h_gate/h = [ 0.6, 0.6, 0.6, ..., 0.6, 0.6, 0.6 ]
       - potassium_channel/potassium_channel_n_gate/n = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
    - Rates:
       - membrane/V/prime = [ 0.6, 0.2, 0.1, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - membrane/i_Stim = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/alpha_m = [ 0.2, 0.2, 0.2, ..., 0.2, 0.2, 0.2 ]
       - sodium_channel/sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/potassium_channel_n_gate/alpha_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/i_Na = [ 1.0, 1.1, 1.1, ..., 1.2, 1.2, 1.2 ]
       - sodium_channel/sodium_channel_m_gate/beta_m = [ 4.0, 4.1, 4.1, ..., 4.0, 4.0, 4.0 ]
       - sodium_channel/sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/beta_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/i_K = [ -4.8, -4.6, -4.4, ..., -4.4, -4.4, -4.4 ]
       - leakage_current/i_L = [ 3.2, 3.3, 3.3, ..., 3.2, 3.2, 3.2 ]
 - Runge-Kutta (2nd order):
    - Constants:
       - membrane/E_R = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - membrane/Cm = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
       - sodium_channel/g_Na = [ 120.0, 120.0, 120.0, ..., 120.0, 120.0, 120.0 ]
       - potassium_channel/g_K = [ 36.0, 36.0, 36.0, ..., 36.0, 36.0, 36.0 ]
       - leakage_current/g_L = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel/E_Na = [ -115.0, -115.0, -115.0, ..., -115.0, -115.0, -115.0 ]
       - potassium_channel/E_K = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - leakage_current/E_L = [ -10.6, -10.6, -10.6, ..., -10.6, -10.6, -10.6 ]
    - States:
       - membrane/V = [ 0.0, 0.3, 0.4, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/sodium_channel_h_gate/h = [ 0.6, 0.6, 0.6, ..., 0.6, 0.6, 0.6 ]
       - potassium_channel/potassium_channel_n_gate/n = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
    - Rates:
       - membrane/V/prime = [ 0.6, 0.2, 0.1, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - membrane/i_Stim = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/alpha_m = [ 0.2, 0.2, 0.2, ..., 0.2, 0.2, 0.2 ]
       - sodium_channel/sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/potassium_channel_n_gate/alpha_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/i_Na = [ 1.0, 1.1, 1.1, ..., 1.2, 1.2, 1.2 ]
       - sodium_channel/sodium_channel_m_gate/beta_m = [ 4.0, 4.1, 4.1, ..., 4.0, 4.0, 4.0 ]
       - sodium_channel/sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/beta_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/i_K = [ -4.8, -4.6, -4.4, ..., -4.4, -4.4, -4.4 ]
       - leakage_current/i_L = [ 3.2, 3.3, 3.3, ..., 3.2, 3.2, 3.2 ]
 - Runge-Kutta (4th order):
    - Constants:
       - membrane/E_R = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - membrane/Cm = [ 1.0, 1.0, 1.0, ..., 1.0, 1.0, 1.0 ]
       - sodium_channel/g_Na = [ 120.0, 120.0, 120.0, ..., 120.0, 120.0, 120.0 ]
       - potassium_channel/g_K = [ 36.0, 36.0, 36.0, ..., 36.0, 36.0, 36.0 ]
       - leakage_current/g_L = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
       - sodium_channel/E_Na = [ -115.0, -115.0, -115.0, ..., -115.0, -115.0, -115.0 ]
       - potassium_channel/E_K = [ 12.0, 12.0, 12.0, ..., 12.0, 12.0, 12.0 ]
       - leakage_current/E_L = [ -10.6, -10.6, -10.6, ..., -10.6, -10.6, -10.6 ]
    - States:
       - membrane/V = [ 0.0, 0.3, 0.4, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/sodium_channel_h_gate/h = [ 0.6, 0.6, 0.6, ..., 0.6, 0.6, 0.6 ]
       - potassium_channel/potassium_channel_n_gate/n = [ 0.3, 0.3, 0.3, ..., 0.3, 0.3, 0.3 ]
    - Rates:
       - membrane/V/prime = [ 0.6, 0.2, 0.1, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/m/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_h_gate/h/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/n/prime = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
    - Algebraic:
       - membrane/i_Stim = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - sodium_channel/sodium_channel_m_gate/alpha_m = [ 0.2, 0.2, 0.2, ..., 0.2, 0.2, 0.2 ]
       - sodium_channel/sodium_channel_h_gate/alpha_h = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/potassium_channel_n_gate/alpha_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - sodium_channel/i_Na = [ 1.0, 1.1, 1.1, ..., 1.2, 1.2, 1.2 ]
       - sodium_channel/sodium_channel_m_gate/beta_m = [ 4.0, 4.1, 4.1, ..., 4.0, 4.0, 4.0 ]
       - sodium_channel/sodium_channel_h_gate/beta_h = [ 0.0, 0.0, 0.0, ..., 0.0, 0.0, 0.0 ]
       - potassium_channel/potassium_channel_n_gate/beta_n = [ 0.1, 0.1, 0.1, ..., 0.1, 0.1, 0.1 ]
       - potassium_channel/i_K = [ -4.8, -4.6, -4.4, ..., -4.4, -4.4, -4.4 ]
       - leakage_current/i_L = [ 3.2, 3.3, 3.3, ..., 3.2, 3.2, 3.2 ]
